### PR TITLE
feat: improvement to rpc package

### PR DESCRIPTION
**Pull request by Khaomi**
**Please describe the changes this PR makes and why it should be merged:**

**Status and versioning classification:**

### CODE REVIEW

**Review by OfficialSirH — APPROVED** [review]
Looking pretty epicc if I do say so myself

File: packages/rpc/src/client.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-import { APIUser, OAuth2Scopes, RESTPostOAuth2ClientCredentialsResult, Routes, Snowflake } from 'discord-api-types/v10';
+import { setTimeout, clearTimeout } from 'node:timers';
+import { URLSearchParams } from 'node:url';
+import type { APIUser, OAuth2Scopes, RESTPostOAuth2ClientCredentialsResult, Snowflake } from 'discord-api-types/v10';
+import { Routes } from 'discord-api-types/v10';
 import { fetch } from 'undici';
+// eslint-disable-next-line import/extensions
+import { RPCError } from './error';
+// eslint-disable-next-line import/extensions
 import { transports } from './transports';
```
Comment (almeidx) [inline packages/rpc/src/client.ts]: Can't you just do this?
```suggestion
import { RPCError } from './error.js';
import { transports } from './transports/index.js';
```

File: packages/rpc/src/client.ts [hunk 2]
```diff
@@ -27,46 +33,68 @@ import type {
...
-import { RPCCommands, RPCEvents, type LobbyType, RelationshipType } from './typings/types';
-import { pid as getPid, uuid } from './util';
-
-function subKey(event: string, args?: unknown[]) {
-	return `${event}${JSON.stringify(args)}`;
-}
+// eslint-disable-next-line import/extensions
+import { RPCCommands, RPCEvents, type LobbyType, RelationshipType, RPCErrorCodes } from './typings/types';
+// eslint-disable-next-line import/extensions
+import { pid as getPid } from './util';
```
Comment (almeidx) [inline packages/rpc/src/client.ts]: Can't comment on deleted lines 🙄 
```suggestion
import { RPCCommands, RPCEvents, type LobbyType, RelationshipType, RPCErrorCodes } from './typings/types.js';
import { pid as getPid } from './util.js';
```

File: packages/rpc/src/error.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// eslint-disable-next-line import/extensions
+import { RPCErrorCodes } from './typings/types';
```
Comment (almeidx) [inline packages/rpc/src/error.ts]:
```suggestion
import { RPCErrorCodes } from './typings/types.js';
```

File: packages/rpc/src/index.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
-import { RPCClient } from './client';
+// eslint-disable-next-line import/extensions
 import { register as registerUtil } from './util';
```
Comment (almeidx) [inline packages/rpc/src/index.ts]:
```suggestion
import { register as registerUtil } from './util.js';
```

File: packages/rpc/src/index.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
-	// eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call
-	registerUtil(`discord-${id}`);
-// @ts-expect-error
+export const register = (id: string) => registerUtil(`discord-${id}`);
+
+// @ts-expect-error - TS don't know that window can sometimes exist
 export const browser = typeof window !== 'undefined';
+
+// eslint-disable-next-line import/extensions
+export { RPCClient as Client } from './client';
```
Comment (almeidx) [inline packages/rpc/src/index.ts]:
```suggestion
export { RPCClient as Client } from './client.js';
```

File: packages/rpc/src/transports/index.ts [hunk 6]
```diff
@@ -1,11 +1,13 @@
...
-	connect: () => Promise<void> | void;
-	close: () => Promise<void>;
-	ping: () => void;
-	send: (data: any) => void;
-}
-
+// eslint-disable-next-line import/extensions
 import { default as ipc } from './ipc';
+// eslint-disable-next-line import/extensions
 import { WebSocketTransport as websocket } from './websocket';
```
Comment (almeidx) [inline packages/rpc/src/transports/index.ts]:
```suggestion
import { default as ipc } from './ipc.js';
import { WebSocketTransport as websocket } from './websocket.js';
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Comment by Khaomi** [comment]
this was caused by me being bad at git, sorry.

**Pull request merged**